mantidproject/mantid
# Fix CompareWorkspaces for weighted events.

**Description of work.**
The code didn't handle the detailed comparison of weighted events. Now it does.

**To test:**
Run this code:
```
ws1 = CreateSampleWorkspace('Event')
ws2 = CreateSampleWorkspace('Event')
ws1 *= 1.1
ws2 *= 1.2
CompareWorkspaces(ws1, ws2)
```
Shouldn't crash. Check the message in the output workspace.

If you know how to create workspaces with `WeightedEventNoTime` events then those should be tested as well.

Fixes #23330.

*This does not require release notes* because the change is only relevant for developers.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
